sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/prometheus/MicrometerProducerListener.java

```diff
@@ -0,0 +1,60 @@
+package org.sdase.commons.server.kafka.prometheus;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.binder.kafka.KafkaClientMetrics;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.kafka.clients.producer.Producer;
+
+/** Listener adds or removes Kafka internal metrics for each producer to a micrometer registry. */
+public class MicrometerProducerListener {
```

Comment:
Do we want to move this class one level up and make it package private provide to not expose it as public API.